python/dolfinx/mesh.py

```diff
@@ -121,19 +121,53 @@ def Mesh(comm, cell_type, x, cells, ghosts, degree=1, ghost_mode=cpp.mesh.GhostM
     return create_mesh(comm, cells, x, domain, ghost_mode)
 
 
-def MeshTags(mesh, dim, indices, values):
+class MeshTags(object):
+    def __init__(self, *args):
+
+        if len(args) == 4:
+            mesh, dim, indices, values = args
+
+            if isinstance(values, int):
+                values = numpy.full(indices.shape, values, dtype=numpy.intc)
+            elif isinstance(values, float):
+                values = numpy.full(indices.shape, values, dtype=numpy.double)
+
+            dtype = values.dtype.type
+            if dtype not in _meshtags_types.keys():
+                raise KeyError("Datatype {} of values array not recognised".format(dtype))
+
+            fn = _meshtags_types[dtype]
+            self._cpp_object = fn(mesh, dim, indices, values)
+            self.values = self._cpp_object.values
+            self.indices = self._cpp_object.indices
```

Comment:
else: raise?

Reply:
No. else pass. The constructor is called with no arguments in the classmethod.

Reply:
I'm not very happy with this design, but I am not sure how to do it better.